feature/OAR-18 - CRUD for Race

Danielpazm98 opened the pull request:
CRUD for race. May need a few modifications

Danielpazm98 committed:
feat(roleplay view): update of race create view and form

all tests passing to create view and form; template also working.

Danielpazm98 committed:
feat(roleplay view): update of race create view and form

all tests passing to create view and form; template also working.

Danielpazm98 committed:
test(test): test passes for createview

Danielpazm98 committed:
fix(formatting): flake8 format for new functions

Danielpazm98 committed:
Merge

Danielpazm98 committed:
feat(views): update view and delete view works

update view and delete view for Race model now work as intended, with the associated forms and layouts.

Danielpazm98 committed:
mend

Danielpazm98 committed:
feat(race view): list view for race

Danielpazm98 committed:
feat(view): race list view

race list view update

LeCuay committed:
Release v1.1 (#87)

* fix(urls): add `debug_toolbar` under `if statement (#85)

Fixed error on importing `debug_toolbar` for non development
environments so added under if statement.
Also removed old `.env.example` variables.

BREAKING CHANGE: Added security by `ALLOWED_HOST` on WebSockets.

Changes to be committed:
	modified:   .env.example
	modified:   oilandrope/asgi.py
	modified:   oilandrope/urls.py

* fix(ci): update `codecov` action to `v2`

There was a depreaction on `codecov/codecov@v1` so we moved to `v2`.
Also added codecov system on `development` branches.

Changes to be committed:
	modified:   .github/workflows/django.yml
	modified:   .github/workflows/django_dev.yml

* fix(ci): syntax incorrect

We cannot use `[]`.

Changes to be committed:
	modified:   .github/workflows/django.yml
	modified:   .github/workflows/django_dev.yml

* fix(translations): some translations didn't work

Because of how `gettext_lazy` and `gettext` work some translations were
incorrectly displayed.
This (along other things related to translations) has been fixed.

Changes to be committed:
	modified:   common/enums.py
	modified:   common/files/utils.py
	modified:   dynamic_menu/enums.py
	modified:   locale/es/LC_MESSAGES/django.po
	new file:   registration/migrations/0007_alter_user_options.py
	modified:   registration/models.py
	modified:   registration/templates/allauth/socialaccount/connections.html
	modified:   registration/templates/registration/password_change.html
	modified:   registration/templates/registration/password_reset.html
	modified:   registration/templates/registration/password_reset_confirm.html
	modified:   registration/templates/registration/resend_email.html
	modified:   roleplay/enums.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/templates/roleplay/place/place_create.html
	modified:   roleplay/templates/roleplay/world/world_confirm_delete.html
	modified:   roleplay/templates/roleplay/world/world_create.html
	modified:   roleplay/views.py

* test(html): add test coverage for templates

Now Django Templates (either `.html` or `.txt`) will be encapsulated in
coverage to check if every condition is being triggered.

BREAKING CHANGES: Added `DEBUG_TEMPLATE` variable and now is needed.

Changes to be committed:
	modified:   .env.example
	modified:   .github/workflows/django.yml
	modified:   bot/bot.py
	deleted:    bot/templates/bot/templates/place_list.txt
	deleted:    chat/templates/chat/index.html
	modified:   common/utils/env.py
	modified:   oilandrope/settings.py
	modified:   poetry.lock
	modified:   pyproject.toml
	deleted:    roleplay/templates/roleplay/music/music_list.html
	modified:   setup.cfg
	modified:   tests/api/serializers/test_roleplay_serializer.py
	modified:   tests/bot/test_bot.py
	modified:   tests/chat/test_models.py
	modified:   tests/registration/test_forms.py

* fix(test): increase on coverage

Increased coverage by testing certain templates from third-party
packages since these templates were being overriden.

Changes to be committed:
	modified:   oilandrope/dev_settings.py
	deleted:    registration/templates/allauth/socialaccount/login_cancelled.html
	deleted:    registration/templates/allauth/socialaccount/signup.html
	new file:   tests/allauth/__init__.py
	new file:   tests/allauth/test_views.py

* feature/OAR-35 - CRUD for Session (#86)

* feat(roleplay): obligatory chat

There has been created a migration in order to make `Chat`obligatory.
This is intended to be approached by first creating the chat and then
adding it to the session.

Changes to be committed:
	modified:   roleplay/admin.py
	new file:   roleplay/migrations/0007_alter_playerinsession_options_and_more.py
	modified:   roleplay/models.py
	modified:   tests/roleplay/test_models.py

* feat(form): add form for `Session`

Added form for `Session` that works with `HtmlThreadMail` and added
tests for checking data and that mails are being sent.

Changes to be committed:
	modified:   common/forms/widgets.py
	modified:   common/tools/mail.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/models.py
	modified:   tests/roleplay/test_forms.py

* feat(views): add `SessionCreateView`

Base logic for creating a Session is done. Missing tests.

Changes to be committed:
	modified:   oar_email/templates/email_templates/invitation_email.html
	modified:   roleplay/admin.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/forms/layout.py
	modified:   roleplay/templates/roleplay/session/session_create.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* test(views): add tests for `SessionCreateView`

`SessionCreateView` is now tested against required fields, email
invitations sending and data being created correctly.

Changes to be committed:
	modified:   roleplay/baker_recipes.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/models.py
	modified:   roleplay/views.py
	modified:   tests/api/test_views.py
	modified:   tests/roleplay/test_models.py
	modified:   tests/roleplay/test_views.py

* fix(viewsets): `SessionViewSet` works better now

Fixed a permission problem and added tests for actions.

Changes to be committed:
	modified:   api/permissions/roleplay.py
	modified:   api/routers/roleplay.py
	modified:   api/serializers/common.py
	modified:   api/serializers/roleplay.py
	modified:   api/viewsets/mixins.py
	modified:   api/viewsets/roleplay.py
	modified:   roleplay/baker_recipes.py
	modified:   roleplay/templates/roleplay/include/world_card.html
	modified:   tests/api/serializers/test_roleplay_serializer.py
	modified:   tests/api/viewsets/test_roleplay_viewsets.py

* feat(api): add `StaffListAllMixin`

The new Mixin allows admins to retrieve all objects from a `QuerySet`.
This is intended to give admins a possibility to search a complete list
of objects.

Changes to be committed:
	modified:   api/viewsets/mixins.py
	modified:   api/viewsets/roleplay.py
	modified:   tests/api/viewsets/test_roleplay_viewsets.py

* feat(views): world is given to `SessionCreateView`

In order to not overwhelm the user with the amount of worlds, we set
initial the given one.

Changes to be committed:
	modified:   roleplay/baker_recipes.py
	modified:   roleplay/templates/roleplay/include/world_card.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* fix(views): mail invitations now are in `views.py`

It didn't make sense for the form to send invitations. What if we reuse
this form? Email invitations are still being sent.

In `SessionCreateView` makes more sense to be.

Changes to be committed:
	modified:   oar_email/templates/email_templates/invitation_email.html
	modified:   roleplay/admin.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/templates/roleplay/place/place_detail.html
	modified:   roleplay/templates/roleplay/session/session_create.html
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_forms.py

* feat(views): add `SessionDetailView`

Added view to see a particular session.

Changes to be committed:
	modified:   roleplay/templates/roleplay/session/session_detail.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* feat(views): add view for deleting Session

Added `SessionDeleteView` and test for checking user is a `game_master`
otherwise it gave 403.

Changes to be committed:
	modified:   oilandrope/settings.py
	new file:   roleplay/templates/roleplay/session/session_confirm_delete.html
	modified:   roleplay/templates/roleplay/session/session_detail.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* feat(views): add view for editting session

Added `SessionUpdateView` and tests for checking updates can be only
performed by a game master.

Changes to be committed:
	modified:   roleplay/forms/layout.py
	modified:   roleplay/templates/roleplay/session/session_detail.html
	new file:   roleplay/templates/roleplay/session/session_update.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* feat(views): add view for listing sessions

Added `SessionListView` in order to show the player all of their
sessions.

Also added `Sessions` to menu.

Changes to be committed:
	modified:   core/templates/core/includes/menu.html
	modified:   roleplay/models.py
	modified:   roleplay/templates/roleplay/session/session_confirm_delete.html
	modified:   roleplay/templates/roleplay/session/session_create.html
	modified:   roleplay/templates/roleplay/session/session_detail.html
	new file:   roleplay/templates/roleplay/session/session_list.html
	modified:   roleplay/templates/roleplay/session/session_update.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* feat(template): style on session listing template

Added some styles for `SessionListView` and added a new field `image` on
session so users can set whatever cover they want.

Changes to be committed:
	modified:   roleplay/admin.py
	modified:   roleplay/migrations/0007_alter_playerinsession_options_and_more.py
	modified:   roleplay/models.py
	modified:   roleplay/templates/roleplay/session/session_list.html
	modified:   roleplay/views.py

* fix(env): add environment variable

Added environment vairable for coverage on `DEBUG_TEMPLATE`.

Changes to be committed:
	modified:   .github/workflows/django_dev.yml

* fix(test): increase coverage on listing

Added some tests to increase coverage on `session_list.html`.

Changes to be committed:
	modified:   roleplay/templates/roleplay/session/session_list.html
	modified:   tests/roleplay/test_views.py

* feat(views): add view for joining a session

Added `SessionJoinView` and test in order to check if the given string
is a correct format.

Changes to be committed:
	modified:   common/tools/__init__.py
	new file:   common/tools/crypt.py
	modified:   oar_email/templates/email_templates/invitation_email.html
	modified:   oilandrope/settings.py
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	new file:   tests/common/tools/test_crypt.py
	modified:   tests/roleplay/test_views.py

* fix(views): remove `extra_tags`

Since `settings.MESSAGES_TAGS` changed we don't need this anymore.

Changes to be committed:
	modified:   registration/views.py
	modified:   tests/registration/test_views.py

* refactor(code): extractsend session mail to func

Since `SessionUpdateView` followed the same logic as `SessionCreteView`
it didn't really make sense to repeat this logic but to make a separated
function for this.
Also refactored some style on HTML.

BREAKING CHANGES: Renamed `world_confirm_delete.html` to
`place_confirm_delete.html` since it made more sense because is used by
all `Place`.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/forms/layout.py
	modified:   roleplay/templates/roleplay/include/world_card.html
	renamed:    roleplay/templates/roleplay/world/world_confirm_delete.html -> roleplay/templates/roleplay/place/place_confirm_delete.html
	modified:   roleplay/templates/roleplay/place/place_detail.html
	modified:   roleplay/templates/roleplay/session/session_detail.html
	modified:   roleplay/templates/roleplay/world/world_list.html
	modified:   roleplay/urls.py
	new file:   roleplay/utils/__init__.py
	new file:   roleplay/utils/invitations.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* refactor(ui): change HTML to be more user-friendly

Changed `session_list.html` to have a better presentation and also added
`session_list_item.html` in order to have a more legible HTML.

Changes to be committed:
	new file:   roleplay/templates/roleplay/session/include/session_list_item.html
	modified:   roleplay/templates/roleplay/session/session_detail.html
	modified:   roleplay/templates/roleplay/session/session_list.html
	modified:   roleplay/views.py

* feat(lan): translations

Added new translations.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/templates/roleplay/place/place_detail.html

* fix(models): more length on chat

Added more length on `chat.models.Chat.name` since tests were failing
because of varying char 50 was too short.

Changes to be committed:
	new file:   chat/migrations/0007_alter_chat_name.py
	modified:   chat/models.py
	new file:   dynamic_menu/migrations/0005_alter_dynamicmenu_menu_type.py

* fix(html): new tab on tableto open

Quick fix so the user doesn't have to leave our website.

Changes to be committed:
	modified:   roleplay/templates/roleplay/session/include/session_list_item.html

* fix(dependencies): problem with `werkzeug`

Werkzeug removed `environ['...'.shutdown` on version `2.1` so
`runserver_plus`.

Changes to be committed:
	modified:   poetry.lock
	modified:   pyproject.toml

Danielpazm98 committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

Danielpazm98 committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

Danielpazm98 committed:
feat(test): tests now work for race views

Danielpazm98 committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

Danielpazm98 committed:
Signed-off-by: Daniel Paz Marcos <danielpaz4c@gmail.com>

fix(test): race update test fix

Danielpazm98 committed:
Signed-off-by: Daniel Paz Marcos <danielpaz4c@gmail.com>

fix(lint): lint fix to views and test_views

Danielpazm98 committed:
fix(isort): fix isort on files

Danielpazm98 committed:
feat(test): finish test for views and templates

dependabot[bot] committed:
build(deps): bump django from 4.0.3 to 4.0.4 (#90)

Bumps [django](https://github.com/django/django) from 4.0.3 to 4.0.4.
- [Release notes](https://github.com/django/django/releases)
- [Commits](https://github.com/django/django/compare/4.0.3...4.0.4)

---
updated-dependencies:
- dependency-name: django
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LeCuay committed:
Merge branch 'develop'

LeCuay committed:
build(dep): update `Twistted`

Updated `Twistted` from `22.2` to `24.2`.

Changes to be committed:
	modified:   poetry.lock
	modified:   pyproject.toml

LeCuay committed:
Merge branch 'develop'

Danielpazm98 committed:
test(races): race tests update

includes tests for detailview

Danielpazm98 committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

Danielpazm98 committed:
feat(menu): race now shows in upper menu

Danielpazm98 committed:
fix(tmp_files): remove temp files

LeCuay commented:
Seems like you don't have translations, please remember to execute `python manage.py makemessages` in order to generate `.po` files and then translate them.

- [ ] Add new translations

LeCuay committed:
Merge branch 'develop'

Danielpazm98 committed:
feat(translation): translation to race crud

LeCuay commented on locale/es/LC_MESSAGES/django.po:
```diff
@@ -1870,6 +1893,18 @@ msgstr "opciones"
+#, fuzzy
+#| msgid "Affected by armor"
+msgid "Affected by Armor"
+msgstr "afectado por la armadura"
```
Change "Affected" to "affected by armor" to maintain lowercase

- [ ] Change to lowercase

LeCuay commented on roleplay/templates/roleplay/race/race_update.html:
```diff
@@ -0,0 +1,25 @@
+  </main>
+{% endblock body_content %}
+
+{% block javascript %}
```
If not needed then delete

Danielpazm98 committed:
refactor(change request): general improvement from pull request review

Danielpazm98 committed:
refactor(recipe): improvement to baker recipes

Danielpazm98 committed:
refactor(pull request change): improve code quality

Danielpazm98 committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

Danielpazm98 committed:
fix(view): fix to success_url

Danielpazm98 committed:
feat(translation):  race translation

Danielpazm98 committed:
feat(template):  race list change

LeCuay committed:
Merge remote-tracking branch 'origin' into feature/OAR-18

LeCuay committed:
Merge branch 'develop' into feature/OAR-18

LeCuay committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

LeCuay committed:
feat(OAR-19): [WIP] Working on OAR-18

Changes to be committed:
	new file:   core/templates/core/includes/menus/unauthenticated_menu.html
	modified:   core/templates/core/includes/menus/worldbuilding_menu.html
	modified:   roleplay/models.py
	modified:   roleplay/templates/roleplay/race/race_list.html

LeCuay committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

LeCuay committed:
lan(OAR-18): add spanish translations

Updated `django.po` for translations.
Also fixed problem on tests.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   tests/roleplay/test_views.py

LeCuay committed:
feat(OAR-18): menu reestructure

Moved plain HTML to individual files.

Changes to be committed:
	new file:   core/templates/core/includes/menus/world_building/races_menu.html
	new file:   core/templates/core/includes/menus/world_building/worlds_menu.html
	modified:   core/templates/core/includes/menus/worldbuilding_menu.html

LeCuay committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

LeCuay committed:
feat(OAR-18): reestructre race list page

Added and moved some components in order to improve UX/UI.
Also added a filter form to improve navigation.

Changes to be committed:
	modified:   common/templates/common/include/filter_accordion.html
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/filters/__init__.py
	modified:   roleplay/filters/filters.py
	modified:   roleplay/templates/roleplay/race/race_list.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

LeCuay committed:
feat(OAR-18): generic model for images

Created a generic model for handling multiple images to different
instances starting by `roleplay.Race`.

Changes to be committed:
	modified:   common/constants/models.py
	new file:   common/migrations/0006_alter_vote_content_type_image_and_more.py
	modified:   common/models.py

LeCuay committed:
Merge branch 'develop' of github.com:oil-rope/oil-and-rope into feature/OAR-18

LeCuay committed:
feature/OAR-112 - List Races View & Create on Admin site (#110)

* feat(OAR-18): model declaration

Created migrations for the new `Image` model with generic relation to
any object that needs a set of images.
Also reestructure the `Race` model to meet Acceptance Criteria.

Changes to be committed:
	new file:   common/migrations/0006_alter_vote_content_type.py
	renamed:    common/migrations/0006_alter_vote_content_type_image_and_more.py -> common/migrations/0007_image_image_common_imag_content_4fea9c_idx.py
	modified:   common/models.py
	new file:   registration/migrations/0009_alter_profile_language.py
	modified:   registration/models.py
	modified:   roleplay/baker_recipes.py
	new file:   roleplay/migrations/0012_remove_race_image_remove_race_users_race_campaign_and_more.py
	modified:   roleplay/models.py
	modified:   tests/bot/test_models.py
	modified:   tests/common/test_models.py
	modified:   tests/registration/test_models.py
	modified:   tests/roleplay/test_models.py

* feat(OAR-112): WIP on admin

Changes to be committed:
	modified:   common/admin.py
	modified:   oilandrope/dev_settings.py
	modified:   oilandrope/settings.py
	modified:   roleplay/admin.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/views.py
	modified:   tests/api/serializers/test_roleplay_serializers.py
	modified:   tests/roleplay/test_forms.py
	modified:   tests/roleplay/test_models.py
	modified:   tests/roleplay/test_views.py

* feat(OAR-112): admin site

Finished develop for admin site on `Race` model.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   oilandrope/urls.py
	modified:   roleplay/admin.py
	modified:   roleplay/models.py
	modified:   tests/roleplay/test_models.py

* feat(OAR-112): add trait to `Race`

Added `Trait` and `TratiType` so we can associated them to races.
TODO: We'll need to add this new models to admin.

Changes to be committed:
	modified:   common/constants/models.py
	modified:   common/filters/mixins.py
	modified:   common/models.py
	modified:   oilandrope/settings.py
	modified:   poetry.lock
	modified:   pyproject.toml
	modified:   roleplay/baker_recipes.py
	new file:   roleplay/filters/enums.py
	modified:   roleplay/filters/filters.py
	new file:   roleplay/migrations/0013_traittype_trait.py
	new file:   roleplay/migrations/0014_populate_initial_trait_types.py
	modified:   roleplay/models.py
	modified:   roleplay/templates/roleplay/race/race_list.html
	new file:   roleplay/translation.py
	modified:   tests/roleplay/test_models.py

* feat(OAR-112): finish admin site

Added admin site for managing `Race`, `Trait` and `TraitType`.
Added translations.

Changes to be committed:
	modified:   common/admin.py
	modified:   common/templates/errors/404.html
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   oar_email/templates/email_templates/campaign_join_request.html
	modified:   registration/templates/api/request_token.html
	modified:   registration/views.py
	modified:   roleplay/admin.py
	modified:   roleplay/migrations/0013_traittype_trait.py
	modified:   roleplay/models.py
	modified:   roleplay/templates/roleplay/race/race_list.html

* test(OAR-112): bypass for `fail-under`

**IMPORTANT**

This will be removed once `OAR-112` is implemented.

Changes to be committed:
	modified:   .github/workflows/django.yml

* fix(OAR-112): test fixes

Changes to be committed:
	modified:   tests/registration/test_views.py
	modified:   tests/roleplay/test_models.py

* ci(OAR-112): bypass for test coverage

Changes to be committed:
	modified:   .github/workflows/django.yml

* test(OAR-112): testing `RaceListView`

Finished testing for `RaceListView`.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/baker_recipes.py
	modified:   roleplay/filters/filters.py
	modified:   roleplay/migrations/0013_traittype_trait.py
	modified:   roleplay/models.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py
	modified:   tests/utils.py

* feat(OAR-112): display for `RaceListView`

Now races are shown on a cards and pagination.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/baker_recipes.py
	modified:   roleplay/templates/roleplay/include/race_card.html
	modified:   roleplay/templates/roleplay/race/race_list.html

* ci(OAR-112): bypass no longer needed

Removed bypass for uncovered files.

Changes to be committed:
	modified:   .github/workflows/django.yml
diff --git a/.github/workflows/django.yml b/.github/workflows/django.yml
index 20dc313..fffbe29 100644
--- a/.github/workflows/django.yml
+++ b/.github/workflows/django.yml
@@ -98,8 +98,8 @@ jobs:
         # TODO: OAR-18 fix the `fail-under` to remove this bypass
         run: |
           poetry run coverage run ./manage.py test
-          poetry run coverage report --skip-covered --fail-under=90
-          poetry run coverage xml --fail-under=90
+          poetry run coverage report --skip-covered
+          poetry run coverage xml

       - name: Update coverage
         uses: codecov/codecov-action@v3

* test(OAR-112): image testing

Test for checking default values on `description` and test to
check if `images` are displayed.

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* feat(OAR-112): styling

Modified display on race cards.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/templates/roleplay/include/race_card.html
	modified:   roleplay/templates/roleplay/race/race_list.html
        modified:   tests/roleplay/test_views.py

* feat(OAR-112): styles missing

Some minor fixes on styling.

Changes to be committed:
	modified:   docker/docker-entrypoint.sh
	modified:   roleplay/templates/roleplay/race/race_list.html

* test(OAR-112): fixing because of truncations

Since we truncate the description it no longers appears full.

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* fix(OAR-112): redirection fix

Redirection field by `next` QueryParam wasn't being taken into
account.

Changes to be committed:
        modified:   locale/es/LC_MESSAGES/django.po
	modified:   registration/forms/forms.py
	modified:   registration/urls.py
	modified:   registration/views.py
	modified:   tests/roleplay/test_views.py

LeCuay committed:
feature/OAR-113 - Capability Races for Place and Campaign (#113)

* feat(OAR-113): logic for form

Created basic logic in order to create a race with 3 optional images.

Changes to be committed:
	modified:   common/forms/__init__.py
	new file:   common/forms/forms.py
	modified:   roleplay/forms/__init__.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/forms/layout.py
	modified:   roleplay/templates/roleplay/race/race_create.html
	modified:   roleplay/templates/roleplay/race/race_list.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py

* feat(OAR-113): showing ability modifiers

Now on `RaceListView` we should the ability modifiers alongside with the
name so the user can have a quick review of how they work.

Changes to be committed:
	modified:   roleplay/models.py
	modified:   roleplay/templates/roleplay/include/race_card.html

* fix(OAR-113): clarification

Added a help text to tell the user that they can upload images and the
amount.
Also removed the asterisk from the label so it fits the UX/UI.

Changes to be committed:
	modified:   common/forms/forms.py
	modified:   roleplay/templates/roleplay/race/race_create.html

* test(OAR-113): `RacePlaceForm` tests

Added tests for the new form.

Changes to be committed:
	modified:   roleplay/forms/forms.py
	new file:   roleplay/migrations/0015_alter_race_affected_by_armor_alter_race_charisma_and_more.py
	modified:   roleplay/models.py
	modified:   tests/roleplay/test_forms.py

* dep(OAR-113): update dependencies

Updated dependencies and dropped `dpytest` since it doesn't seem to work
anymore and project it's almost abandoned.

Changes to be committed:
	modified:   poetry.lock
	modified:   pyproject.toml
	modified:   pytest.ini
	modified:   roleplay/models.py
	modified:   setup.cfg
	deleted:    tests/bot/cogs/__init__.py
	deleted:    tests/bot/cogs/test_misc.py
	deleted:    tests/bot/cogs/test_rol.py
	deleted:    tests/bot/conftest.py
	deleted:    tests/bot/test_bot.py

* feat(OAR-113): test for Generic Formset

FormSets are really hard to deal with... Tests clarified a little about
what we can and what we can't do.

Changes to be committed:
	modified:   .coveragerc
	modified:   common/files/upload.py
	modified:   common/forms/forms.py
	modified:   common/models.py
	modified:   oilandrope/runner.py
	modified:   poetry.lock
	modified:   pyproject.toml
	modified:   pytest.ini
	modified:   roleplay/urls.py
	new file:   tests/common/forms/__init__.py
	new file:   tests/common/forms/test_forms.py
	modified:   tests/common/test_models.py
	modified:   tests/roleplay/test_forms.py
	modified:   tests/roleplay/test_views.py

* feat(OAR-113): test for the new CRUD

Working `RaceCreateForPlaceView` and tests.

Changes to be committed:
	modified:   common/baker_recipes.py
	modified:   core/context_processors.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/templates/roleplay/place/place_detail.html
	modified:   roleplay/templates/roleplay/race/race_create.html
	modified:   roleplay/templates/roleplay/race/race_list.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/common/forms/test_forms.py
	modified:   tests/roleplay/test_views.py

* feat(OAR-113): translations

Added translations.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/templates/roleplay/race/race_create.html

* dep(OAR-113): fix dependencies

`djangorestframework` is failing at NullBooleanField.

Changes to be committed:
	modified:   poetry.lock
	modified:   pyproject.toml

* fix(OAR-113): failing tests

There were some failing tests.

Changes to be committed:
	modified:   tests/roleplay/test_forms.py
	modified:   tests/roleplay/test_views.py

* test(OAR-113): test with url with place

New tests with the alternative URL that includes place.

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* feat(OAR-113): minor fixes on formset

Setup ImageFormSet for adding multiple images to a Race.

Changes to be committed:
	new file:   common/files/__init__.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/templates/roleplay/race/race_create.html
	modified:   roleplay/views.py

* feat(manage): read environment variables as files (#111)

Added the capability to read some specific environment variables as
files.

Changes to be committed:
	modified:   common/utils/env.py
	modified:   manage.py
	modified:   tests/common/utils/test_env.py

* hotfix/github-actions-migration (#112)

* feat(manage): read environment variables as files

Added the capability to read some specific environment variables as
files.

Changes to be committed:
	modified:   common/utils/env.py
	modified:   manage.py
	modified:   tests/common/utils/test_env.py

* ci(migration): no longer using `set-output`

Migrating to using environment variables as recommended by GitHub
Actions.

Changes to be committed:
	modified:   .github/workflows/django.yml

* fix(deploy): missing loading secrets on asgi

We setup the loading secrets for `manage.py` but not for `asgi.py`
neither `gunicorn_conf.py`.

Changes to be committed:
	modified:   oilandrope/asgi.py
	modified:   oilandrope/gunicorn_conf.py

* ci(sha): adding image as SHA

Images labeled by SHA.

Changes to be committed:
	modified:   .github/workflows/django.yml

* fix(ci): invalid input

Changes to be committed:
	modified:   .github/workflows/django.yml

* fix(ci): trying breakline separator

Changes to be committed:
	modified:   .github/workflows/django.yml

* fix(ci): changing env var format

Changes to be committed:
	modified:   .github/workflows/django.yml

* fix(ci) using github object

Changes to be committed:
	modified:   .github/workflows/django.yml

* feat(OAR-113): race creation for campaigns

Implemented `RaceCreateForCampaignView` for linking races directly to
campaigns.
Also created `ImageFormsetMixin` for dealing with instances with
multiple images.

Changes to be committed:
	modified:   common/forms/widgets.py
	new file:   common/views/mixins.py
	modified:   registration/models.py
	modified:   roleplay/forms/__init__.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/forms/layout.py
	modified:   roleplay/templates/roleplay/race/race_list.html
	modified:   roleplay/urls.py
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

* fix(OAR-113): failing tests

Fixed failing tests on `TestCampaignUpdateView`.

Changes to be committed:
	modified:   .coveragerc
	modified:   tests/roleplay/test_views.py

* lan(OAR-113): ammend translations

Since there are new lines we need to translate them.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po

* dep(OAR-113): missing daphne

For some reason we were missing `daphne`.

Changes to be committed:
	modified:   poetry.lock
	modified:   pyproject.toml

* feat(OAR-113): better information display

Added more user-friendly information for when a User doesn't have the
capability to create a race.
Also minor fixes.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   registration/models.py
	modified:   roleplay/templates/roleplay/race/race_create.html
	modified:   roleplay/templates/roleplay/race/race_list.html
	modified:   roleplay/views.py
	modified:   tests/roleplay/test_views.py

LeCuay committed:
feature/OAR-114 - Delete Race View (#114)

* feat(OAR-114): implementation of deletion of race

Created a new Mixin View (`DeletedObjectsView`) that will show what
objects will be deleted when the given objects is deleted.

Changes to be committed:
	modified:   common/templates/common/include/breadcrumb.html
	new file:   common/templates/common/include/to_delete_accordion.html
	new file:   common/templates/common/layout/base_confirm_delete.html
	modified:   common/templatetags/string_utils.py
	new file:   common/views/edit.py
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/templates/roleplay/include/race_card.html
	deleted:    roleplay/templates/roleplay/race/race_confirm_delete.html
	modified:   roleplay/views.py
	modified:   tests/common/templatetags/test_string_utils.py
	new file:   tests/common/views/test_edit.py
	modified:   tests/roleplay/test_views.py
	modified:   tests/utils.py

* test(OAR-114): missing tests on utils

Added tests in order to cover everything under `roleplay.utils`.

Changes to be committed:
	modified:   common/utils/faker.py
	modified:   roleplay/utils/invitations.py
	new file:   tests/roleplay/utils/test_invitations.py

* refactor(OAR-114): removing unnecessary funcs

`Place` had a lot of methods that weren't being used.

Changes to be committed:
	modified:   roleplay/models.py

* feat(OAR-114): removing relationships from str

Since `__str__` method is used for representation it shouldn't include
relationships since it can obfuscate the database.

Changes to be committed:
	modified:   common/models.py
	modified:   tests/common/test_models.py

* feat(OAR-114): fix on the deletion mixin class

Fixes on `DeletedObjectsView` mixin and added the feature of
`get_format_callback` to customize how items are displayed on the delete
view.

Changes to be committed:
	modified:   common/templates/common/include/to_delete_accordion.html
	modified:   common/templates/common/layout/base_confirm_delete.html
	modified:   common/views/edit.py
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/templates/roleplay/place/place_confirm_delete.html
	modified:   roleplay/views.py
	modified:   tests/common/views/test_edit.py
	modified:   tests/roleplay/test_views.py

* lan(OAR-114): fix translations

Minor fix on the translations.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po

LeCuay committed:
fix(OAR-18): duplication on campaign

Due to union instead on `unique` a duplication was retrieved on
campaigns.

Changes to be committed:
	modified:   registration/models.py

LeCuay committed:
fix(OAR-18): now *using* distinct

Seems like ^ didn't do the trick.

Changes to be committed:
	modified:   registration/models.py

LeCuay committed:
feature/OAR-116 - Update Race View (#115)

* refactor(OAR-116): serializers are not tackled

Within the scope of this task serializers are not going to be used
neither created.

Changes to be committed:
	modified:   api/serializers/roleplay.py

* refactor(OAR-116): minor reestructuring

Changes to be committed:
	modified:   common/views/json.py
	modified:   core/static/js/setupUtils.js
	modified:   tests/api/serializers/test_roleplay_serializers.py

* feat(OAR-116): implementation of updating races

Created a `ImageFormsetMixin` for managing `common.Image` generic model
in both scenarios creating and updating.

Changes to be committed:
	modified:   common/views/mixins.py
	new file:   core/types.py
	modified:   locale/es/LC_MESSAGES/django.po
	modified:   roleplay/forms/__init__.py
	modified:   roleplay/forms/forms.py
	modified:   roleplay/forms/layout.py
	modified:   roleplay/templates/roleplay/include/race_card.html
	modified:   roleplay/templates/roleplay/race/race_update.html
	modified:   roleplay/views.py
	new file:   tests/common/views/test_mixins.py

* feat(OAR-116): translatiosn fix

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po

* test(views): minimal testing for the editting view

Added tests to `RaceUpdateView` that can be found on
`TestRaceUpdateView`.

Changes to be committed:
	modified:   .coveragerc
	modified:   core/templates/core/includes/menu.html
	deleted:    core/templates/core/includes/menus/unauthenticated_menu.html
	modified:   poetry.lock
	modified:   pyproject.toml
	modified:   roleplay/forms/layout.py
	modified:   tests/roleplay/test_views.py

* build(dependencies): poetry config

Amend on the poetry config files.

hanges to be committed:
	modified:   pyproject.toml

* build(dependencies): update Django to >=4.2.0

Now we are up to Django 4.2!

Changes to be committed:
	modified:   poetry.lock
	modified:   pyproject.toml

* test(OAR-116): fix on missing coverage

The view `PasswordResetConfirmView` wasn't accessing
`registration/password_reset_confirm.html` but instead to Django Admin
template.

Changes to be committed:
	modified:   .coveragerc
	renamed:    registration/templates/registration/password_reset_confirm.html -> registration/templates/registration/password_reset_confirm_no_admin.html
	modified:   registration/views.py
	modified:   tests/registration/test_views.py

* test(OAR-116): data tests on `RaceUpdateView`

Added tests for valid and invalid data on `TestRaceUpdateView`. Missing
formset tests for multiple images.

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* test(OAR-116): all tests for `RaceUpdateView`

Created all necessary tests for the `RaceUpdateView`.

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* test(OAR-116): missing coverage lines

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* fix(OAR-116): ignoring uncoverable files

Some files are just impossible or very difficult to cover so we just
ignore them.
Also added more coverage for `place_detail.html`.

Changes to be committed:
	modified:   .coveragerc
	modified:   tests/roleplay/test_views.py

* fix(OAR-116): fix iSort

Changes to be committed:
	modified:   tests/roleplay/test_views.py

* fix(OAR-116): translation fix

Minor relocation of translations.

Changes to be committed:
	modified:   locale/es/LC_MESSAGES/django.po
